### PR TITLE
Add support for HostInterface and HostInterfaceItem (4.4 Enhancement)

### PR DESCRIPTION
**Pull request by mergwyn**
#### Pull Request (PR) description
Add two new parameters HostInterface and HostInterfaceItem to the agent introduced in 4.4

#### This Pull Request (PR) fixes the following issues
Enhancement - N/A

### CODE REVIEW

**Comment by baurmatt** [comment]
Hey @mergwyn, thanks for you PR! Look great! :)

Could you please squash your commits?

**Comment by mergwyn** [comment]
I've done that, hopefully correctly!

File: manifests/agent.pp [hunk 1]
```diff
@@ -250,6 +258,8 @@
   $hostnameitem                                   = $zabbix::params::agent_hostnameitem,
   $hostmetadata                                   = $zabbix::params::agent_hostmetadata,
   $hostmetadataitem                               = $zabbix::params::agent_hostmetadataitem,
+  $hostinterface                                  = $zabbix::params::agent_hostinterface,
+  $hostinterfaceitem                              = $zabbix::params::agent_hostinterfaceitem,
```
Comment (baurmatt) [inline manifests/agent.pp]: Can you please add data types for both parameters?

File: spec/classes/agent_spec.rb [hunk 2]
```diff
@@ -122,6 +122,28 @@
...
+        end
+
+        it { is_expected.to contain_file(config_path).with_content %r{^HostInterfaceItem=system.hostname$} }
+      end
+
+      context 'configuration file with hostinterface' do
+        let :params do
+          {
+            hostinterface: 'testinterface',
+            hostinterfaceitem: 'system.hostname'
```
Comment (baurmatt) [inline spec/classes/agent_spec.rb]: Can we only test `hostinterface` here? `hostinterfaceitem` is already tested above. At least as far as I can see, there is no required connection between both parameters.

**Comment by baurmatt** [comment]
> I've done that, hopefully correctly!

Looks good, thanks! :) Good you please address both comments I've just added? After that I will happily merge this change.

**Comment by mergwyn** [comment]
Changes made as request - I also fixed a lint error about a very long line 201 in agent.pp in passing.

**Comment by baurmatt** [comment]
@mergwyn Looks like somethings gone wrong :( Can you please check the failing tests and the second commit? :)

**Comment by mergwyn** [comment]
May bad - I forgot about 'undef'. If the build passes I'm happy to squash these last two commits. Would you prefer everything in a single commit, or one for the original change and one for the comments?

**Comment by baurmatt** [comment]
> Would you prefer everything in a single commit, or one for the original change and one for the comments?

I prefer one commit as this makes the git history cleaner :)

**Comment by mergwyn** [comment]
Looks good now, though not sure if the Add tablespace parameter commit that crept in when I squashed commits is an issue.
Thanks for your patience.

File: manifests/database.pp [hunk 3]
```diff
@@ -129,20 +132,27 @@
   $database_host_ip                = $zabbix::params::server_database_host_ip,
   $database_charset                = $zabbix::params::server_database_charset,
   $database_collate                = $zabbix::params::server_database_collate,
+  Optional[String[1]] $database_tablespace = $zabbix::params::server_database_tablespace,
```
Comment (bastelfreak) [inline manifests/database.pp]: is this change in the PR on purpose or by accident? the tablespace isn't related to the hostinterface/hostinterfaceitem?

**Comment by mergwyn** [comment]
Hi, I accidentally included that commit when swashing my changes, and I now I don't know how to remove it. Sorry!

**Comment by mergwyn** [comment]
Hi, I've decided the simplest course of action is to close this PR and take a clean fork, reapply the changes and create a new PR!

**Pull request closed without merging**